### ISSUE TITLE
Seq2Seq Decoder not working

### ISSUE THREAD

**Issue by jhillhouse92** [aws/aws-neuron-sdk#379]
I'm following https://awsdocs-neuron.readthedocs-hosted.com/en/latest/src/examples/pytorch/transformers-marianmt.html for implementing a Seq2Seq model (T5-Large). There are some issues:

1. `model.model` property didn't exist
```
class NeuronEncoder(torch.nn.Module):

    def __init__(self, model):
        super().__init__()
        self.encoder = model.model.encoder

    def forward(self, input_ids, attention_mask):
        return self.encoder(input_ids, attention_mask=attention_mask, return_dict=False)


class NeuronDecoder(torch.nn.Module):

    def __init__(self, model, max_length):
        super().__init__()
        self.weight = model.model.shared.weight.clone().detach()
        self.bias = model.final_logits_bias.clone().detach()
        self.decoder = model.model.decoder
```

2. `final_logits_bias` didn't exist

```
class NeuronDecoder(torch.nn.Module):

    def __init__(self, model, max_length):
        super().__init__()
        self.weight = model.model.shared.weight.clone().detach()
        self.bias = model.final_logits_bias.clone().detach()
        self.decoder = model.model.decoder
```

3. `encoder_padding_mask`, `decoder_padding_mask`, and `decoder_causal_mask` didn't exist

```
# Invoke the decoder
        hidden, = self.decoder(
            input_ids=input_ids,
            encoder_hidden_states=encoder_outputs,
            encoder_padding_mask=attention_mask,
            decoder_padding_mask=None,
            decoder_causal_mask=causal_mask,
            return_dict=False,
            use_cache=False,
        )
```

After removing all of that and increasing memory up to 100GB+, neuroncc finally compiled in 2+ hours. However, the decoder model generates an error when trying to load the decoder model:
```
2021-Dec-27 18:11:33.0461 29553:29553 ERROR  TDRV:dma_ring_allocate                       No more pages for ring type: 1
2021-Dec-27 18:11:33.0461 29553:29553 ERROR  TDRV:kbl_model_add                           create_data_refill_rings() error
2021-Dec-27 18:11:33.0538 29553:29553 ERROR  NMGR:dlr_kelf_stage                          Failed to load subgraph
2021-Dec-27 18:11:33.0538 29553:29553 ERROR  NMGR:stage_kelf_models                       Failed to stage graph: kelf-a.json to NeuronCore
2021-Dec-27 18:11:33.0545 29553:29553 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: 1.7.3.0+c02f9fde4-/tmp/tmpoei5grmf, err: 4
Traceback (most recent call last):
  File "test.py", line 27, in <module>
    model_neuron.decoder = torch.jit.load(os.path.join(directory, 'decoder.pt'))
  File "/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch_neuron/jit_load_wrapper.py", line 13, in wrapper
    script_module = jit_load(*args, **kwargs)
  File "/home/ubuntu/anaconda3/envs/aws_neuron_pytorch_p36/lib/python3.6/site-packages/torch/jit/_serialization.py", line 161, in load
    cpp_module = torch._C.import_ir_module(cu, str(f), map_location, _extra_files)
RuntimeError: Could not load the model status=4 message=Allocation Failure
```

Other issues report this as a compiler issue but it fails at a different place in the other issues (`forward` method as opposed to TorchScript `import_ir_module` method). In looking at other TorchScript modules, it seems this might be related to different PyTorch versions used in tracing vs loading. Both versions were using `torch-1.9.1` and `torch-neuron-1.9.1`.  

I'm recompiling now with torch-neuron-1.7.1 and torch-1.7.1, but wasn't sure if that would be compatible with Neuron Runtime 2.x.

**Comment by shebbur-aws** [comment]
Hi @jhillhouse92,

Glad you were able to get through compilation issues with original trace. Yes, using latest torch-neuron 1.7.1.* would be recommended here and should be compatible with Neuron Runtime 2.x .
Regarding the load issue you are seeing, the fix for it will be available in one of our upcoming releases.

Thanks,
Shruthi.

**Comment by awsrjh** [comment]
Hi

This has been resolved in the recent 1.17.0 [release](https://github.com/aws/aws-neuron-sdk/releases/tag/v1.17.0). 

Please find more information here: 

https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/index.html#neuron-1-17-0-01-20-2022
